### PR TITLE
write_map() should write its last entry with a newline

### PR DESCRIPTION
**Pull request by mcovarr**
Fixes to address WomMap's `write_map()` / `tsvSerialize()` lack of terminal newline. WomObject already appears to be handling this correctly ([here](https://github.com/broadinstitute/cromwell/blob/develop/wom/src/main/scala/wom/values/WomObject.scala#L119) and [here](https://github.com/broadinstitute/cromwell/blob/develop/wom/src/main/scala/wom/values/WomObject.scala#L125)), and WomArray also [looks okay](https://github.com/broadinstitute/cromwell/blob/develop/wom/src/main/scala/wom/values/WomArray.scala#L88).

### CODE REVIEW

File: wom/src/main/scala/wom/values/WomMap.scala [hunk 1]
```diff
@@ -70,7 +70,7 @@ final case class WomMap private(womType: WomMapType, value: Map[WomValue, WomVal
   def tsvSerialize: Try[String] = {
     (womType.keyType, womType.valueType) match {
       case (_: WomPrimitiveType, _: WomPrimitiveType) =>
-        Success(value.map({case (k, v) => s"${k.valueString}\t${v.valueString}"}).mkString("\n"))
+        Success(value.map({case (k, v) => s"${k.valueString}\t${v.valueString}"}).mkString(start="", sep="\n", end="\n"))
```
Comment (cjllanwarne) [inline wom/src/main/scala/wom/values/WomMap.scala:73]: Per WDL 1.1, the best source we have, `If the Map is empty, an empty file is written.`, which I don't think is the case here.

File: centaur/src/main/resources/standardTestCases/read_write_map/read_write_map.wdl [hunk 2]
```diff
@@ -26,10 +26,30 @@ task read_map {
   }
 }
 
+task assert_write_map_terminal_newline {
```
Comment (cjllanwarne) [inline centaur/src/main/resources/standardTestCases/read_write_map/read_write_map.wdl:29]: This feels like it would be better, and easier to add more edge cases to, as a unit test of the `def tsvSerialize: Try[String]` than here an integration test

Comment (jgainerdewar) [reply centaur/src/main/resources/standardTestCases/read_write_map/read_write_map.wdl:29]: +1

File: wom/src/main/scala/wom/values/WomMap.scala (hunk 1 above)
Comment (mcovarr) [reply wom/src/main/scala/wom/values/WomMap.scala:79]: Updated. I noticed the 1.1 spec defines `write_map()` as a function for `Map[String, String]` **only**, though the code and tests here should work for all primitive types.

**Review by jgainerdewar — APPROVED** [review]
Thanks for jumping on this!
